beesdoo_base/models/membercard.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from random import randint
+import uuid
+
+class MemberCard(models.Model):
+    
+    def _get_current_user(self):
+        return self.env.uid
+    
+    def _get_current_client(self):
+        # TODO : this does not work
+        return self.env['res.partner'].search([('id', '=',self.env.context['active_id'])])
+    
+    def _compute_bar_code(self):
+        rule = self.env['barcode.rule'].search([('name', '=', 'Customer Barcodes')])[0]
+        nomenclature = self.env['barcode.nomenclature']
+        size = 13-len(rule.pattern)
+        ean = rule.pattern + str(uuid.uuid4().fields[-1])[:size] #str(randint(10**(size-1), 10**size-1))
+        code = ean[0:12] + str(nomenclature.ean_checksum(ean))
```

Comment:
0:12 should be useless otherwise we have a problem since we are making a checksum on the full ean..